more fourth wall stuff

MIslam92 opened the pull request:
### **What is the context of this PR?**

Added new features for fourth-wall including ability to add amount of approvals required in URL and number of commits are displayed on each card now.

### **How to review**

Run it and see it it displays what it is meant to display and if the configurable url works i.e. approvals=x in Url as a tag.
https://trello.com/c/pYR4vYdS

shahi645 reviewed (approved):
Just delete the icons from img folder, but otherwise works perfectly 👍🏿

MIslam92 committed:
Added feature for no. of commits and configurable URL

Include no. of commits for each PR and ability to configure required amount of approvals via URL. Also, small UI changes.

adoublebarrel commented on javascript/core.js:
```diff
@@ -148,4 +148,10 @@
 
+  FourthWall.minimumApprovals = FourthWall.getQueryVariable('approvals');
+
+  if( typeof FourthWall.minimumApprovals === 'undefined'){
```
Better to set the default value and then override; more readable when you have more default values and it's the more common pattern in most code bases.

adoublebarrel commented on fourth-wall.css:
```diff
@@ -29,14 +29,18 @@ body {
 }
 
+.holder{
+    padding-bottom: 40px;
```
Is there a reason for the mix of px and rems in padding/margins/etc.?

adoublebarrel commented on fourth-wall.css:
```diff
@@ -64,20 +83,18 @@ body {
 
 .base-sync {
-  padding: 0.15rem 0.45rem;
+   padding: 0.5rem 0.75rem;
```
See px vs rem comment above.

jonnyshaw89 reviewed (approved):
Other than what @adoublebarrel said, all looks good.
He's just really fussy! :)

adoublebarrel commented:
Thanks Jon!

adoublebarrel commented on javascript/core.js:
```diff
@@ -148,6 +148,7 @@
   FourthWall.showLabels = FourthWall.getQueryVariable('labels') === 'true';
 
+  FourthWall.minimumApprovals = 2;
   FourthWall.minimumApprovals = FourthWall.getQueryVariable('approvals');
```
So this change actually makes the code worse. The new line at 151 is completely worthless without changing the rest of the code that follows it. I appreciate this has turned into a bit of a slog but I'd like to see a little more thought go into these changes please. Feel free to comment back or Slack me if I'm not being clear enough.

In this case you could just switch the logic of the following `if` to only update `minimumApprovals` if `'approvals'` is *not* undefined. Alternatively you could 'or' the default value with the call to `getQueryVariable` assuming you're confident that it will always return a falsy value or a valid int. Happy to talk this through if you'd find it helpful.

adoublebarrel commented on fourth-wall.css:
```diff
@@ -38,7 +38,7 @@ body {
 
 .holder{
-    padding-bottom: 40px;
+    padding-bottom: 2.5rem;
```
My previous comment was querying the inconsistency rather than asking you to make the change. This file still has a mixture of `px` and `rem` values for margins and padding after this change so the situation isn't any better and I have no more info on why there's a mix than I did yesterday.

The reason I'm asking is that there's often a relationship between `font-size` and the amount of whitespace in a design. So if you make a change to the font size you often have to update the whitespace (margins/padding/line heights/etc.). Intelligent use of `em's` and `rem's` can mean that you just make one change to a font-size rather than font size + all the whitespace rules.

If the mixture of `px` and `rem` units hasn't got much thought behind it we should put some work on the backlog to address it to make the Fourth Wall less expensive to change.

MIslam92 replied:
Thanks for the feedback @adoublebarrel very help and makes things a lot clearer I'll sort out the inconsistencies and the logic behind the configurable URL minimum approvals as well 👍

adoublebarrel reviewed (approved):
Good to go.
